devito/operator.py

```diff
@@ -124,30 +123,33 @@ def __init__(self, expressions, **kwargs):
         # Finish instantiation
         super(Operator, self).__init__(self.name, nodes, 'int', parameters, ())
 
-    @cached_property
-    def _argument_defaults(self):
+    def _argument_defaults(self, arguments):
         """
         Derive all default values from parameters and ensure uniqueness.
         """
         default_args = ArgumentMap()
         for p in self.input:
-            default_args.update(p.argument_defaults())
+            if p.name not in arguments:
+                default_args.update(p.argument_defaults())
+            else:
+                default_args.update(p.argument_defaults(data=False))
         for p in self.dimensions:
-            if p.is_Sub:
-                default_args.update(p.argument_defaults(default_args))
-        return {k: default_args.reduce(k) for k in default_args}
+            if p.name not in arguments:
```

Comment:
probably don't need this ``if``; if you do, you can still join it with the nested one

Reply:
join the IFs pls